### ISSUE TITLE
Classify same as closest training vector

### ISSUE THREAD

**Issue by muttley2k** [Quantronauts/quantum_data-classifier#1]
Let the classifier essentially contain all the training vectors, and during testing the classifier's task should be to figure out which training vector is the closest (e.g. in terms of vector distance) to the vector being classified. The test vector will be assigned the same label as the training vector closest to it.

The philosophy behind is that if we want to classify regions of the Hilbert space, we cannot really be smarter than the training set... that's all the information we have. (Actually, SVMs are based on a similar idea, but smarter :-)

I expect that the one-shot restriction during testing will make things tricky.

**Comment by mickahell** [comment]
See that : https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm

**Comment by muttley2k** [comment]
Yeah, something like this, but quantum :-)

**Comment by Zed-Is-Dead** [comment]
This is also somehow linked to kernel methods (cf. Maria's paper): the training data determines the quantum model.
